packages/react-mutation-mapper/src/component/track/ExonTrack.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { observer } from 'mobx-react';
+import { computed, makeObservable } from 'mobx';
+
+import { OncoKbFilterValue } from '../../filter/OncoKbFilter';
+import { DataFilterType } from '../../model/DataFilter';
+import MutationMapperStore from '../../model/MutationMapperStore';
+import { default as Track, TrackProps } from './Track';
+import { TrackItemSpec, TrackItemType } from './TrackItem';
+import { Mutation } from 'cbioportal-utils';
+import { EnsemblTranscript } from '../../../../genome-nexus-ts-api-client/dist';
+
+type ExonTrackProps = TrackProps & {
+    store: MutationMapperStore<Mutation>;
+};
+
+type ExonDatum = { rank: number; length: number; start: number };
+
+@observer
+export default class ExonTrack extends React.Component<ExonTrackProps, {}> {
+    constructor(props: any) {
+        super(props);
+        makeObservable(this);
+    }
+
+    @computed get transcriptId(): string | undefined {
+        return this.props.store.activeTranscript?.result;
+    }
+
+    @computed get transcript(): EnsemblTranscript | undefined {
+        return this.transcriptId
+            ? this.props.store.transcriptsByTranscriptId[this.transcriptId]
+            : undefined;
+    }
+
+    @computed get exonSpecs(): TrackItemSpec[] {
+        if (!this.transcriptId || !this.transcript) {
+            return [];
+        }
+
+        const exonInfo = this.exonInformation;
+
+        const altColors = ['#007FFF', '#35BAF6'];
+
+        return exonInfo.map((exon: ExonDatum, index: number) => {
+            const startCodon = exon.start;
+            const endCodon = exon.start + exon.length;
+            const exonLength = exon.length;
+            const isSkippable = Number.isInteger(exonLength);
+
+            return {
+                color: altColors[index % 2],
+                startCodon: startCodon,
+                endCodon: endCodon,
+                label: exon.rank.toString(),
+                labelColor: '#FFFFFF',
+                tooltip: (
+                    <span>
+                        <h1> Exon {exon.rank} </h1>
+                        <br></br>
+                        Start: {startCodon.toFixed(2).toString()}
+                        <br></br>
+                        End: {endCodon.toFixed(2).toString()}
+                        <br></br>
+                        Exon length (amino acid):{' '}
+                        {exonLength.toFixed(2).toString()}
+                        <br></br>
+                        Exon {isSkippable ? 'is' : "isn't"} a multiple of 3
+                    </span>
+                ),
+            };
+        });
+    }
+
+    @computed get exonInformation(): ExonDatum[] {
```

Comment:
There is still some custom logic in this function. It might be better to have utility function(s) with a few unit tests for this logic.